### ISSUE TITLE
Decrease accuracy after upgrade from v0.11 to v0.12

### ISSUE THREAD

**Issue by edgarriba** [tensorflow/tensorflow#7160]
Hi! I've recently updated from v0.11 to v.012 using pip with CUDA 8.0, cuDNN 5.1in a TitanX under Ubuntu 16.04 LTS.

During the training of a small CNN triplet network with triplet margin loss I noticed that the performance in terms of accuracy went down by a factor of ~40%.  Getting around 9% and 14% FPR95 with v0.11 and v0.12 respectively, both at epoch 30th. Notice with v0.12 the accuracy gets stuck in early epoch 10th.

The optimizer that I'm using is Momentum with a LR of 1e-4 without decay policy.
I've also added the regularization term with a weight decay of 1e-4.

Did you introduce any change with last versions in the optimization API that could make this difference?

PD: I left here the link to the code.
https://github.com/vbalnt/tfeat/tree/master/tensorflow
Notice that we are trying to reproduce from the original LuaTorch version.

**Comment by rasbt** [comment]
Hi,
I have some continuous integration tests set up for some of my projects, where I run the code on small training sets and didn't notice any change in performance when I was switching from 0.11 to 0.12. The integration tests are running on CPU though. Maybe could you check for the first epochs if you get the same results in 0.11 vs 0.12? This would tell us that there is something going on with the GPU stuff maybe. Also, do you maybe have any random gens that are not seeded?

**Comment by yaroslavvb** [comment]
@edgarriba Ouch, those ones are hard to troubleshoot, can you try with latest nightly from https://github.com/tensorflow/tensorflow ?

**Comment by edgarriba** [comment]
Hi guys, sorry for this late reply. @rasbt you were right I finally found that opencv resizes routine has a random component. So, after setting all the random seeds the results in different versions are almost the same.

@yaroslavvb Besides, and not sure if it's the correct place to ask, but as I mentioned above we have been struggling to replicate some results from a paper we have using triplets networks. The original code is in LuaTorch, and during the conversion of the loss function to TF we are not getting the same performance in terms of accuracy.

If someone want to check here is the code
https://github.com/vbalnt/tfeat/blob/master/tensorflow/train/tfeat/loss.py#L13

**Comment by yaroslavvb** [comment]
@edgarriba there's no magic bullet for matching performance of two frameworks unfortunately. Last time I tried to match Torch's l-BFGS in TensorFlow, I ended up saving results after each operation in both Torch and in TensorFlow, and comparing them numerically one-by-one until I found the difference. See [SaveTensor1D](https://github.com/yaroslavvb/lbfgs/blob/master/mnist_sgd.lua#L91) and acompanying [notebook]( https://github.com/yaroslavvb/lbfgs/blob/master/Porting%20Torch%20l-BFGS%20to%20immediate.ipynb)

**Comment by yaroslavvb** [comment]
I'll close it for now since this doesn't seem like a bug in TensorFlow

**Comment by edgarriba** [comment]
@yaroslavvb sure I'll try. Thanks!
